### DISCUSSION TITLE
Ideas for NFT on Filecoin!

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/community#102] [Developers]
Exporing how we can bring NFT to filecoin!

**Comment by funktafide** [comment]
I would like to see some NFTs on Filecoin, but beyond that, seems like Filecoin would be a good long term storage option for these NFT marketplaces.  Especially if what was in this twitter thread is true.
 https://twitter.com/jonty/status/1372163423446917122?s=21

**Comment by tim-murmuration** [comment]
**Problem:**

As an end-user of NFT minting & hosting platforms, I have a lack of clear knowledge about where, how, and for how long the underlying digital assets (e.g., image files, etc) associated with my NFTs are stored. If the platform goes out of business, then it's likely my assets will be lost. This problem potentially affects both me as a seller and my buyers. 

**Note:**

It's becoming increasingly common to suggest assets be stored on IPFS to mitigate these issues. However, the same problem still arises that if an IPFS node hosting my assets goes down (and no one else is hosting it redundantly), then those assets also will become inaccessible. 

**Proposed Solution:** 

As an end user of NFT platforms, I would like the ability (as either a seller or buyer) to indicate at time of minting or purchase that I would additionally like to buy persistent long term storage for the underlying assets on the Filecoin network. This would include redundant distributed backups, and to whatever extent made possible by the technology "guarantees" that the assets will be stored and retrievable for a given period of time (which would be renewable). 

Ideally, this integrated solution would be blockchain-agnostic, and could function equally well alongside any other NFT-enabled blockchain (e.g., Ethereum, Cardano, Tezos, EOSIO, etc.) and on any NFT platform.

Additionally, Protocol Labs should consider the feasibility of developing a native-to-Filecoin NFT asset class, where these storage and retrieval costs are automatically built into the minting & sale processes, and are explicitly available for inspection.

**Reply by tim-murmuration** [reply]
Filecoin network fees could be explicitly customized (e.g., time period, geo-location, compliance regimes, etc.), including who covers the associated fees (seller, buyer, or shared cost), etc.

**Reply by eike-heimpel** [reply]
I was thinking about stuff along the same line, googled a bit and found this thread. 

One additional thing I thought would be useful is for NFT owners to be able to give read-only permissions for the underlying assets to NFT platforms (marketplaces, showrooms, your friends website) where the platform can read the contents of the asset directly from source in a transparent and provable manner. This access could be limited to a certain time and revoked at any time.

**Comment by tim-murmuration** [comment]
The other thing I'd like to see is an open-source means for artists to run their own NFT platforms that are self-hosted, so that they don't have to be reliant on third party platforms, where much of the control and features are out of their hands. So this would be like a "WordPress for NFTs" or a "Shopify for NFTs" that would ideally have quick & simple install, plus plenty of customizations. 

I see no reason why a system such as this would not also employ redundant IPFS & Filecoin storage on the back-end.
